sockets/sockets.js

```diff
@@ -1810,7 +1821,11 @@ module.exports = function (io) {
 
 
 
-
+var updateCurrentPlayersList = function () {
```

Comment:
There's two issues with this section. 

1) You are calling function getPlayerUsernamesAndStatusesFromAllSockets() n squared times (line 1826) since you have a for loop in a for loop. With a linear increase in online players(sockets), there will be an exponential increase in processing time which isn't good. Solution can simply to move the function call outside the forEach call, and then pass in the variable each time in the sock.emit(). 

2) There are a lot of updateCurrentPlayerList() function calls (throughout the server sockets.js file). Ideally each client will send a small message to the server whenever they change their status, and then update it in an object/array on the server side to then be re-emitted to everyone. This will save a lot of time checking over players who haven't done anything... But for now I think it's ok to leave this as it is. 1) is the bigger issue :)